### DISCUSSION TITLE
How do I use `store.startConfiguring()` and `store.finishConfiguring()` ?

### DISCUSSION THREAD

**Discussion by jeremytenjo** [storybookjs/storybook#12050] [Help]
https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#story-store-immutable-outside-of-configuration

The docs are not very clear on how to use `store.startConfiguring()` and `store.finishConfiguring()` .

Where does `store` come from?

**Comment by shilman** [comment]
Those are implementation details, and you probably shouldn't be calling those functions yourself. @tmeasday maybe we should rename those with underscores to ... underscore this point?

**Reply by jeremytenjo** [reply]
Is there a recommended way to load stories dynamically ?

**Comment by joshje** [comment]
For those still wondering how to follow that guidance, I found (by reading source code) that the `storyStore` is available via `window.__STORYBOOK_STORY_STORE__`.

https://github.com/storybookjs/storybook/blob/master/lib/core/src/client/preview/start.ts#L89

```javascript
const storyStore = window.__STORYBOOK_STORY_STORE__;

storyStore.startConfiguring();
...
storyStore.finishConfiguring();
```

**Reply by pyymenta** [reply]
It works for me man! Thank you so much! :tada: :tada: :tada: :tada:

**Reply by pyymenta** [reply]
In my case, I've made a wrapper that encapsulates the start/finish configuring logic into a wrapper:

```js
const configWrapper = (storyLogicFn) => {
  window.__STORYBOOK_STORY_STORE__.startConfiguring();

  storyLogicFn();

  window.__STORYBOOK_STORY_STORE__.finishConfiguring();
};
```

Then, you can use outside config context:

```js
functionThatReturnsAPromise().then(() => {
  configWrapper(() => {
    storiesOf('MyStorybookSection/Component', module)
      .add('some-usecase', () => (
        <MyComponent />
      ));
  });
})
```

And it works like a charm :smile: :smile: :smile:
